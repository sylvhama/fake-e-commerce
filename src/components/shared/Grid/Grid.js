import styled from 'styled-components';
import PropTypes from 'prop-types';

import media from '../../../styled/media';

const repeat = columns => `repeat(${columns},1fr)`;

const Grid = styled.div`
  display: grid;
  grid-template-columns: ${props => repeat(props.columns, props.size)};
  grid-gap: ${props => props.theme.spacing.reg};

  ${media.tablet`
    grid-template-columns: ${props => repeat(props.tabletColumns, props.size)};
    grid-gap: ${props => props.theme.spacing.sm};
  `};
`;

Grid.propTypes = {
  columns: PropTypes.number,
  tabletColumns: PropTypes.number
};

Grid.defaultProps = {
  columns: 5,
  tabletColumns: 2
};

export default Grid;
