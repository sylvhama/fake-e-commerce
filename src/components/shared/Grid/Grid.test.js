import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../tests/helpers';

import Grid from './Grid';

test('Grid renders correctly', () => {
  const rendered = renderWithTheme(<Grid>Grid</Grid>).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('grid-template-columns', 'repeat(5,1fr)');
});

test('Grid renders correctly with 12 columns', () => {
  const rendered = renderWithTheme(<Grid columns={12}>Grid</Grid>).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('grid-template-columns', 'repeat(12,1fr)');
});
