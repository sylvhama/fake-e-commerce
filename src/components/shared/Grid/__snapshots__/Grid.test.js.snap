// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(5,1fr);
  grid-gap: 16px;
}

@media (max-width:48em) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
    grid-gap: 8px;
  }
}

<div
  className="c0"
>
  Grid
</div>
`;

exports[`Grid renders correctly with 12 columns 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  grid-gap: 16px;
}

@media (max-width:48em) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
    grid-gap: 8px;
  }
}

<div
  className="c0"
>
  Grid
</div>
`;
